### PR TITLE
[stlab] Fix cmake config file

### LINKED ISSUE #10343: [stlab] Complains about missing Boost header files
**Host Environment**
- OS: WSL Ubuntu 18.04
- Compiler: gcc 7.4.0 (also tried clang 9)

**To Reproduce**
`./vcpkg install stlab`
Then when you run cmake, it complains about missing boost header files, even though there is no marked boost dependency on windows or linux. Only boost variant on osx.

**Failure logs** 
```
CMake Error at /usr/share/cmake-3.10/Modules/FindBoost.cmake:1947 (message):
  Unable to find the requested Boost libraries.

  Unable to find the Boost header files.  Please set BOOST_ROOT to the root
  directory containing Boost or BOOST_INCLUDEDIR to the directory containing
  Boost's headers.
Call Stack (most recent call first):
  /home/olafurw/vcpkg/scripts/buildsystems/vcpkg.cmake:288 (_find_package)
  /usr/share/cmake-3.10/Modules/CMakeFindDependencyMacro.cmake:48 (find_package)
  /home/olafurw/vcpkg/installed/x64-linux/share/stlab/stlabConfig.cmake:2 (find_dependency)
  /home/olafurw/vcpkg/scripts/buildsystems/vcpkg.cmake:288 (_find_package)
  CMakeLists.txt:12 (find_package)


CMake Error at /home/olafurw/vcpkg/scripts/buildsystems/vcpkg.cmake:288 (_find_package):
  Found package configuration file:

    /home/olafurw/vcpkg/installed/x64-linux/share/stlab/stlabConfig.cmake

  but it set stlab_FOUND to FALSE so package "stlab" is considered to be NOT
  FOUND.  Reason given by package:

  stlab could not be found because dependency Boost could not be found.

Call Stack (most recent call first):
  CMakeLists.txt:12 (find_package)

-- Configuring incomplete, errors occurred!
```

**Additional context**
My best guess is if I install the boost headers it will work. But that's not something I want to install.

### PR DESCRIPTION
**Pull request by PhoebeHui**
Fixes https://github.com/microsoft/vcpkg/issues/10343

Stlab core lib doesn't reply on boost, however 'find_dependency(Boost 1.60.0)'  was added to stlabConfig.cmake file which installed from its source.

On OSX, it still requires boost. https://github.com/stlab/libraries/blob/develop/CMakeLists.txt#L44

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Comment by dan-shaw** [comment]
Would it be possible not to add a version to boost? So that when we update boost in the future, vcpkg will still be able to find it.

File: ports/stlab/portfile.cmake [hunk 1]
```diff
@@ -17,8 +17,10 @@ vcpkg_install_cmake()
...
 vcpkg_copy_pdbs()
 
-# cleanup
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug ${CURRENT_PACKAGES_DIR}/share/cmake)
 
-# handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
\ No newline at end of file
+file(READ ${CURRENT_PACKAGES_DIR}/share/${PORT}/stlabConfig.cmake STLAB_CONFIG)
+string(REPLACE "find_dependency(Boost 1.60.0)" "if(APPLE)\nfind_dependency(Boost 1.73.0)\nendif()" STLAB_CONFIG ${STLAB_CONFIG})
```
Comment (PhoebeHui) [inline ports/stlab/portfile.cmake:23]:
```suggestion
string(REPLACE "find_dependency(Boost 1.60.0)" "if(APPLE)\nfind_dependency(Boost)\nendif()" STLAB_CONFIG ${STLAB_CONFIG})

```

**Pull request merged**